internal/agent/device/config/controller.go

```diff
@@ -126,17 +98,35 @@ func (c *controller) ensureConfigData(ctx context.Context, currentData, desiredD
 
 	// write ignition files to disk and trigger pre hooks
 	c.log.Debug("Writing ignition files")
-	err = c.WriteIgnitionFiles(ctx, desiredIgnition.Storage.Files)
+	err = c.writeIgnitionFiles(ctx, desiredIgnition.Storage.Files)
 	if err != nil {
 		c.log.Warnf("Writing ignition files failed: %+v", err)
 		return fmt.Errorf("failed to apply configuration: %w", err)
 	}
 	return nil
 }
 
-func (c *controller) WriteIgnitionFiles(ctx context.Context, files []ignv3types.File) error {
+// removeObsoleteFiles removes files that are present in the currentFiles but not in the desiredFiles.
+func (c *Controller) removeObsoleteFiles(ctx context.Context, currentFiles, desiredFiles []ignv3types.File) error {
+	removeFiles := computeRemoval(currentFiles, desiredFiles)
+	for _, file := range removeFiles {
+		c.log.Infof("Deleting file: %s", file)
+		// trigger delete pre hook and wait for it to complete
+		c.hookManager.OnBeforeRemove(ctx, file)
+		if err := c.deviceWriter.RemoveFile(file); err != nil {
+			return fmt.Errorf("deleting files failed: %w", err)
```

Comment:
If the file we want to remove doesn't exist is that worth catching and moving forward with?

Reply:
Logically what you're saying makes sense, don't block progress for a successful operation.  RemoveFile is fairly smart and filtering those errors.  SO an error here would be pretty unexpected so I think its valid to report the agent as degraded in this case.

https://github.com/flightctl/flightctl/blob/1016e2c165d34c813cd6b0d60a742d283009399c/internal/agent/device/fileio/writer.go#L63